### PR TITLE
[ENH] extend `scitype` type inference utility to cover `sklearn` metrics and splitters

### PR DESCRIPTION
**Pull request by fkiraly**
This PR extends the `scitype` type inference utility to cover `sklearn` metrics and splitters.

This will later be useful in the benchmarking framework.

### CODE REVIEW

File: sktime/utils/sklearn/_scitype.py [hunk 1]
```diff
@@ -146,3 +190,104 @@ def is_sklearn_clusterer(obj):
     bool, whether obj is an sklearn clusterer
     """
     return is_sklearn_estimator(obj) and sklearn_scitype(obj) == "clusterer"
+
+
+def is_sklearn_splitter(obj):
```
Comment (jgyasu) [inline sktime/utils/sklearn/_scitype.py:196]: I thought all `sklearn` splitters inherit `BaseCrossValidator`?

**Comment by fkiraly** [comment]
why is it running the VM for `SeasonalityPeriodogram`?

File: sktime/utils/sklearn/_scitype.py (hunk 1 above)
Comment (fkiraly) [reply sktime/utils/sklearn/_scitype.py:196]: really? Not when I looked 5 minutes ago.

But please correct me if you think I am wrong.

Comment (jgyasu) [reply sktime/utils/sklearn/_scitype.py:196]: I was looking at this file, [https://github.com/scikit-learn/scikit-learn/blob/c60dae20604f8b9e585fc18a8fa0e0fb50712179/sklearn/model_selection/_split.py](https://github.com/scikit-learn/scikit-learn/blob/c60dae20604f8b9e585fc18a8fa0e0fb50712179/sklearn/model_selection/_split.py)

File: sktime/utils/sklearn/_scitype.py [hunk 2]
```diff
@@ -146,3 +190,104 @@ def is_sklearn_clusterer(obj):
...
+    # 3. Must have correct signature
+    try:
+        sig = signature(obj)
+    except (TypeError, ValueError):
+        return False
+
+    params = list(sig.parameters.values())
+    if len(params) < 2:
+        return False
+    if params[0].name != "y_true" or params[1].name != "y_pred":
```
Comment (jgyasu) [inline sktime/utils/sklearn/_scitype.py]: `params[1].name` can be `"y_proba"` in some cases, e.g., in `brier_score_loss`

**Comment by jgyasu** [comment]
I think this is working since the tests are passing. I have added a comment above, once it is patched, then I think this is ready to merge.

File: sktime/utils/sklearn/_scitype.py (hunk 2 above)
Comment (fkiraly) [reply sktime/utils/sklearn/_scitype.py]: could you suggest a change?

File: sktime/utils/sklearn/_scitype.py (hunk 1 above)
Comment (fkiraly) [reply sktime/utils/sklearn/_scitype.py:196]: I think you are simply wrong, you are generalizing from the first two examples to all?

For example `ShuffleSplit` does not inherit from `BaseCrossValidator`.

File: sktime/utils/sklearn/_scitype.py (hunk 2 above)
Comment (jgyasu) [reply sktime/utils/sklearn/_scitype.py]:
```py
if params[0].name != "y_true" or params[1].name not in {"y_pred", "y_proba"}:
        return False
```

**Comment by fkiraly** [comment]
ok, I have extended functionality to distinguish between deterministic and probabilistic metrics.

I think in benchmarking this will be required, since depending on this `predict` or `predict_proba` needs to be called.

File: sktime/utils/sklearn/_scitype.py (hunk 2 above)
Comment (fkiraly) [reply sktime/utils/sklearn/_scitype.py]: thanks, used that to fix

**Comment by jgyasu** [comment]
> I think in benchmarking this will be required, since depending on this predict or predict_proba needs to be called.

yes, I have encountered it before

**Comment by jgyasu** [comment]
One of the test is failing though

**Review by jgyasu — APPROVED** [review]
Looks great, thanks!
